build-logic/binary-compatibility/src/main/groovy/gradlebuild/binarycompatibility/rules/BinaryBreakingSuperclassChangeRule.groovy

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gradlebuild.binarycompatibility.rules
+
+import japicmp.model.JApiClass
+import japicmp.model.JApiCompatibility
+import javassist.CtClass
+import me.champeau.gradle.japicmp.report.Violation
+
+/**
+ * Workaround for <a href="https://github.com/melix/japicmp-gradle-plugin/issues/56">japicmp issue w.r.t. superclass breakage</a>.
+ *
+ * <p>
+ * Reports simple superclass changes as a breaking change, as it affects what methods can be called with the given type, even if
+ * the methods don't change.
```

Comment:
So if A <-extends- B <-extends- C then a change to C won't count as a breaking change to A, but a change to B would?  Should the entire inheritance tree be compared for shape and compatibility at each node to be truly certain?

Reply:
I will change the doc comment to be clearer, but when I say "simple", I mean _simple_: this is just a fully qualified name check. `japicmp` has already done this work, we're just properly reporting it as a violation instead of ignoring it.

Binary compatibility changes in the superclasses owned by Gradle have always been reported, as `japicmp` reports those on the individual method level, not as part of this superclass object.

Reply:
Got it.  Definitely simple.